diff --git a/qemu_v8/optee_sp_manifest.dts b/qemu_v8/optee_sp_manifest.dts
index 35c2f4b..3cd8b80 100644
--- a/qemu_v8/optee_sp_manifest.dts
+++ b/qemu_v8/optee_sp_manifest.dts
@@ -506,7 +510,7 @@ QEMU_XEN    ?= -drive if=none,file=$(XEN_EXT4),format=raw,id=hd1 \
                   -device virtio-blk-device,drive=hd1
 else
 QEMU_SMP       ?= 2
-QEMU_MEM       ?= 1057
+QEMU_MEM       ?= 4098
 endif
 
 ifeq ($(XEN_BOOT),y)

